torch/_inductor/dependencies.py

```diff
@@ -664,6 +731,82 @@ def extract_loop_body_with_args(
             None, (entry.buffer_name, None, None, None), None, None, None  # type: ignore[arg-type]
         )
     # fn.memory_usage[MemoryUsageType.CHECK_BOUNDS] intentionally skipped
+    # detect indirect broadcast
+    repl_reverse = {v: k for k, v in repl.items()}
+    replacement_mem_dep = {}
+    name_to_node = {node.target: node for node in fn.get_nodes()}
+    for read in inner._reads:
+        if isinstance(read, MemoryDep) and read.replacement:
+            # find out the indirect var in memdep's indexing expr, e.g. indirect0 in
+            # index0 = 4096*indirect0 + p1
+            (
+                additive_symbol,
+                coefficient,
+                multiplied_symbol,
+            ) = extract_coefficient_pattern(read.index)
+            if any(
+                x is None for x in (additive_symbol, coefficient, multiplied_symbol)
+            ):
+                continue
+            # check if the coefficient is the stride
+            buffer = None
+            if read.name in V.graph.name_to_buffer:
+                buffer = V.graph.name_to_buffer[read.name]  # type: ignore[assignment]
+            elif read.name in V.graph.graph_inputs:
+                buffer = V.graph.graph_inputs[read.name]  # type: ignore[assignment]
```

Comment:
Is there an API for this?

Reply:
yes. replace it with try_get_buffer in https://github.com/pytorch/pytorch/pull/145992/commits/abe008be359ee3d7655f4358bf22bd01fe547bdb